app/src/main/java/org/mozilla/fenix/tabstray/TrayPagerAdapter.kt

```diff
@@ -42,12 +47,30 @@ class TrayPagerAdapter(
      */
     private val normalAdapter by lazy {
         ConcatAdapter(
-            InactiveTabsAdapter(context, browserInteractor, interactor, INACTIVE_TABS_FEATURE_NAME, context.settings()),
+            InactiveTabsAdapter(
+                viewLifecycleOwner = lifecycleOwner,
+                tabsTrayStore = tabsTrayStore,
+                tabsTrayInteractor = tabsTrayInteractor,
+                inactiveTabsInteractor = inactiveTabsInteractor,
+                inactiveTabsAutoCloseDialogInteractor = inactiveTabsInteractor,
+                tabDelegate = browserInteractor,
+                featureName = INACTIVE_TABS_FEATURE_NAME,
+            ),
             TabGroupAdapter(context, browserInteractor, tabsTrayStore, TAB_GROUP_FEATURE_NAME),
             TitleHeaderAdapter(),
             BrowserTabsAdapter(context, browserInteractor, tabsTrayStore, TABS_TRAY_FEATURE_NAME)
         )
     }
+
+    private val inactiveTabsInteractor by lazy {
+        DefaultInactiveTabsInteractor(
+            InactiveTabsController(
+                appStore = context.components.appStore,
+                settings = context.settings(),
+            )
+        )
+    }
```

Comment:
Objectively, I agree where you're coming from (this smells code pasta-y haha). I think this just follows the interactor -> controller patter and looks bad here because `DefaultInactiveTabsInteractor` only has the one parameter.